packages/synapse-react-client/src/components/TableColumnSchemaEditor/Validators/BooleanSchema.ts

```diff
@@ -0,0 +1,8 @@
+import { z } from 'zod'
+
+const stringToBoolean = z.enum(['true', 'false']).transform(v => v === 'true')
+
+/**
+ * Validates and converts a value to a valid Synapse BOOLEAN value
+ */
+export const booleanSchema = z.union([stringToBoolean, z.boolean()])
```

Comment:
Accepts only `'true'`, `'false'`, `true`, `false` and when parsed, the returned value is necessarily `true` or `false`